cypress/e2e/room/room-header.spec.ts

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2023 Suguru Hirahara
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import { HomeserverInstance } from "../../plugins/utils/homeserver";
+import { SettingLevel } from "../../../src/settings/SettingLevel";
+
+describe("Room Header", () => {
+    let homeserver: HomeserverInstance;
+
+    beforeEach(() => {
+        cy.startHomeserver("default").then((data) => {
+            homeserver = data;
+            cy.initTestUser(homeserver, "Sakura");
+        });
+    });
+
+    afterEach(() => {
+        cy.stopHomeserver(homeserver);
+    });
+
+    it("should render seven buttons by default", () => {
+        cy.createRoom({ name: "Test Room" }).viewRoomByName("Test Room");
+
+        cy.get(".mx_RoomHeader").within(() => {
+            cy.findAllByRole("button").should("have.length", 7).should("be.visible");
+
+            cy.findByRole("button", { name: "Room options" }).should("be.visible");
+            cy.findByRole("button", { name: "Voice call" }).should("be.visible");
+            cy.findByRole("button", { name: "Video call" }).should("be.visible");
+            cy.findByRole("button", { name: "Search" }).should("be.visible");
+            cy.findByRole("button", { name: "Threads" }).should("be.visible");
+            cy.findByRole("button", { name: "Notifications" }).should("be.visible");
+            cy.findByRole("button", { name: "Room info" }).should("be.visible");
```

Comment:
Basically the object of this test is to ensure that just those seven (not six or eight) buttons are rendered by default. This can be achieved by combining `findAllBy` query with `should(have.length)` by counting the exact number of the all found buttons.

Each `findByRole` query does ensure that button whose name is specified exists, but it cannot ensure that there was not another button rendered besides them unexpectedly.

Based on your hint, https://github.com/matrix-org/matrix-react-sdk/pull/10713/commits/b20ceec2e4d8c3e56321af97be2a890f650d3a28 should optimize the readability and the test flow.